lib/browser_collection.js

```diff
@@ -46,7 +46,24 @@ class BrowserCollection {
     return this.browsers.map((browser) => browser.serialize())
   }
 
-  getResults () {
+  calculateExitCode (results, singleRunBrowserNotCaptured, failOnEmptyTestSuite, failOnFailingTestSuite) {
+    if (results.disconnected || singleRunBrowserNotCaptured) {
+      return 1
+    }
+    if (results.success + results.failed === 0 && !failOnEmptyTestSuite) {
+      return 0
+    }
+    if (results.error) {
+      return 1
+    }
+    if (failOnFailingTestSuite === false) {
+      // No errors found, always return success
```

Comment:
I think the comment would be clearer something like:
```
// Tests executed without infrastructure error, exit with 0 independent of test status.
```
Otherwise we are all tangled up in which fail and which error.

Reply:
Updated comment